modules/ROOT/pages/patterns/shortest-paths.adoc

```diff
@@ -0,0 +1,544 @@
+
+To return all paths that are tied for shortest length, use the keywords `ALL SHORTEST`:
+
+.Query
+[source,cypher]
+----
+MATCH p = ALL SHORTEST (wos)-[:LINK]-+(cnm)
+WHERE hby.name = "Worcester Shrub Hill" AND cnm.name = "Bromsgrove"
+RETURN [n in nodes(p) | n.name] AS stops
+----
+
+.Result
+[role="queryresult",options="header,footer",cols="m"]
+|===
+| stops
+
+| ["Worcester Shrub Hill", "Droitwich Spa", "Bromsgrove"]
+| ["Worcester Shrub Hill", "Worcestershire Parkway Ll", "Bromsgrove"]
+
+1+d|Rows: 2
+
+|===
+
+
+[[shortest-groups]]
+== `SHORTEST k GROUPS`
+
+To return all paths that are tied for first, second, and so on up to the kth shortest length, use `SHORTEST k GROUPS`.
+For example, the following returns the first and second shortest length paths between  `Worcester Shrub Hill` and `Bromsgrove`:
+
+.Query
+[source,cypher]
+----
+MATCH p = SHORTEST 2 GROUPS (wos:Station)-[:LINK]-+(bmv:Station)
+WHERE wos.name = "Worcester Shrub Hill" AND bmv.name = "Bromsgrove"
+RETURN [n in nodes(p) | n.name] AS stops, length(p) AS pathLength
+----
+
+.Result
+[role="queryresult",options="header,footer",cols="2m,m"]
+|===
+| stops | pathLength
+| ["Worcester Shrub Hill", "Droitwich Spa", "Bromsgrove"] | 2
+| ["Worcester Shrub Hill", "Worcestershire Parkway Ll", "Bromsgrove"] | 2
+| ["Worcester Shrub Hill", "Worcester Foregate Street", "Droitwich Spa", "Bromsgrove"] | 3
+| ["Worcester Shrub Hill", "Ashchurch", "Worcestershire Parkway Ll", "Bromsgrove"] | 3
+| ["Worcester Shrub Hill", "Ashchurch", "Cheltenham Spa", "Bromsgrove"] | 3
+2+d|Rows: 5
+|===
+
+The first group includes the two shortest paths with `pathLength = 2` (as seen in the first two rows of the results), and the second group includes the three second shortest paths with `pathLength = 3` (as seen in the last three rows of the results).
+
+If more groups are specified than exist in the graph, only those paths that exist are returned.
+For example, if the paths equal to one of the eight shortest paths are specified for `Worcester Shrub Hill` to `Bromsgrove`, only seven groups are returned:
+
+.Query
+[source,cypher]
+----
+MATCH p = SHORTEST 8 GROUPS (wos:Station)-[:LINK]-+(bmv:Station)
+WHERE wos.name = "Worcester Shrub Hill" AND bmv.name = "Bromsgrove"
+RETURN length(p) AS pathLength, count(*) AS numPaths
+----
+
+.Result
+[role="queryresult",options="header,footer",cols="2*m"]
+|===
+| pathLength | numPaths
+
+| 2 | 2
+| 3 | 3
+| 4 | 1
+| 5 | 4
+| 6 | 8
+| 7 | 10
+| 8 | 6
+
+2+d|Rows: 7
+|===
+
+[[any]]
+== `ANY`
+
+The `ANY` keyword can be used to test the reachability of nodes from a given node(s).
+It returns the same as `SHORTEST 1`, but by using the `ANY` keyword the intent of the query is clearer.
+For example, the following query shows all stations reachable from `Pershore` where the distance between each pair of stations is less than 6 miles:
+
+.Query
+[source,cypher]
+----
+MATCH ANY 
+  (:Station {name: 'Pershore'})-[l:LINK WHERE l.distance < 6]-+(b:Station)
```

Comment:
I would vote for an example where node `b` will be matched to a single node. Partitions are explained only after this example.
A quick look at this doc confused me for a second. I expected to see one result for `ANY`, but saw multiple results (not realizing that any match for `b` would lead to one result).

Reply:
Good point. I've changed the example to be showing the existence of a path between two specific stations that satisfy a constraint.